plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointEntryIterator.java

```diff
@@ -538,6 +546,31 @@ private DeltaLakeTransactionLogEntry buildRemoveEntry(ConnectorSession session,
         return DeltaLakeTransactionLogEntry.removeFileEntry(result);
     }
 
+    private DeltaLakeTransactionLogEntry buildSidecarEntry(ConnectorSession session, int pagePosition, Block block)
+    {
+        log.debug("Building sidecar entry from %s pagePosition %d", block, pagePosition);
+        if (block.isNull(pagePosition)) {
+            return null;
+        }
+        int sidecarFields = 4;
```

Comment:
I don't understand where 4 comes from

Reply:
I think side car has 4 entries 
```
{
  "sidecar":{
    "path": "016ae953-37a9-438e-8683-9a9a4a79a395.parquet",
    "sizeInBytes": 2304522,
    "modificationTime": 1512909768000,
    "tags": {}
  }
}
```

Reply:
Right, 4 comes from field numbers of `sidecar`.